101-vm-user-image-data-disks/azuredeploy.json

```diff
@@ -41,107 +61,125 @@
       }
     },
     "vmSize": {
       "type": "string",
-      "defaultValue": "Standard_A3",
+      "metadata": {
+        "description": "This is the size of your VM"
+      }
+    },
+    "newOrExistingVnet": {
       "allowedValues": [
-        "Standard_A3",
-        "Standard_A4",
-        "Standard_A5",
-        "Standard_A6",
-        "Standard_A7",
-        "Standard_A8",
-        "Standard_A9",
-        "Standard_A10",
-        "Standard_A11",
-        "Standard_D3",
-        "Standard_D4",
-        "Standard_D12",
-        "Standard_D13",
-        "Standard_D14"
+        "new",
+        "existing"
       ],
+      "type": "string",
       "metadata": {
-        "description": "This is the size of your VM"
+        "description": "Select if this template needs a new VNet or will reference an existing VNet"
+      }
+    },
+    "newOrExistingVnetName": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "New or Existing VNet Name"
+      }
+    },
+    "newOrExistingSubnetName": {
+      "type": "string",
+      "metadata": {
+        "description": "New or Existing subnet Name"
+      }
+    },
+    "existingVnetResourceGroupName": {
+      "type": "string",
+      "metadata": {
+        "description": "Resource group of the existing VNET"
       }
     }
   },
   "variables": {
-    "apiVersion": "2015-05-01-preview",
-    "publicIPAddressName": "[concat(parameters('vmName'),'PublicIP')]",
-    "vmName": "[parameters('vmName')]",
-    "virtualNetworkName": "[concat(parameters('vmName'),'VNet')]",
-    "nicName": "[concat(parameters('vmName'),'-nic1')]",
-    "addressPrefix": "10.0.0.0/16",
-    "subnet1Name": "Subnet-1",
-    "subnet1Prefix": "10.0.0.0/24",
+    "imageName": "myCustomImage",
+    "publicIPAddressName": "[concat(parameters('customVmName'),'IP')]",
+    "vmName": "[parameters('customVmName')]",
+    "nicName": "[concat(parameters('customVmName'),'Nic')]",
     "publicIPAddressType": "Dynamic",
-    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
-    "subnet1Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]",
-    "osDiskUrl": "[parameters('osDiskUrlParam')]",
-    "dataDiskUrl": "[parameters('dataDiskUrlParam')]",
-    "storageAccountName": "[take(split(split(parameters('osDiskUrlParam'), '/')[2], '.' ), 0)]",
-    "diagStorageAccountName": "[concat(uniquestring(resourceGroup().id), 'diag')]"
+    "apiVersion": "2015-06-15",
+    "templatelink": "[concat('https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/101-vm-user-image-data-disks/', parameters('newOrExistingVnet'),'vnet.json')]"
   },
   "resources": [
     {
-      "type": "Microsoft.Storage/storageAccounts",
-      "name": "[variables('diagStorageAccountName')]",
-      "apiVersion": "2016-01-01",
+      "type": "Microsoft.Compute/images",
+      "apiVersion": "2016-04-30-preview",
```

Comment:
Can you use the latest api version 2017-03-30 ?

Reply:
done